musl needs LD_LIBRARY_PATH set because when .so is opened with dlopen(/abs/path)
it does not qualify for providing symbols for NEEDED dependency.

--- a/jdk/src/solaris/bin/java_md_solinux.c
+++ b/jdk/src/solaris/bin/java_md_solinux.c
@@ -292,7 +292,7 @@ RequiresSetenv(int wanted, const char *j
     char *dmllp = NULL;
     char *p; /* a utility pointer */
 
-#ifdef AIX
+#if defined(AIX) || (defined(__linux__) && !defined(__GLIBC__))
     /* We always have to set the LIBPATH on AIX because ld doesn't support $ORIGIN. */
     return JNI_TRUE;
 #endif
